File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/merge/model/RelationUpdateItemBuild.java
package de.osthus.ambeth.merge.model;

import java.util.Collection;
import java.util.List;

import de.osthus.ambeth.collections.EmptySet;
import de.osthus.ambeth.collections.HashSet;
import de.osthus.ambeth.collections.ISet;
import de.osthus.ambeth.merge.transfer.RelationUpdateItem;

public class RelationUpdateItemBuild implements IRelationUpdateItem
{
	protected String memberName;

	protected ISet<IObjRef> addedORIs = EmptySet.<IObjRef> emptySet();

	protected ISet<IObjRef> removedORIs = EmptySet.<IObjRef> emptySet();

	public RelationUpdateItemBuild(String memberName)
	{
		this.memberName = memberName;
	}

	@Override
	public String getMemberName()
	{
		return memberName;
	}

	@Override
	public IObjRef[] getAddedORIs()
	{
		if (addedORIs.size() == 0)
		{
			return null;
		}
		return addedORIs.toArray(IObjRef.class);
	}

	@Override
	public IObjRef[] getRemovedORIs()
	{
		if (removedORIs.size() == 0)
		{
			return null;
		}
		return removedORIs.toArray(IObjRef.class);
	}

	public void addObjRef(IObjRef objRef)
	{
		if (addedORIs.size() == 0)
		{
			addedORIs = new HashSet<IObjRef>();
		}
		addedORIs.add(objRef);
	}

	public void addObjRefs(IObjRef[] objRefs)
	{
		for (IObjRef objRef : objRefs)
		{
			addObjRef(objRef);
		}
	}

	public void addObjRefs(List<IObjRef> objRefs)
	{
		for (int a = 0, size = objRefs.size(); a < size; a++)
		{
			addObjRef(objRefs.get(a));
		}
	}

	public void addObjRefs(Collection<IObjRef> objRefs)
	{
		for (IObjRef objRef : objRefs)
		{
			addObjRef(objRef);
		}
	}

	public void removeObjRef(IObjRef objRef)
	{
		if (removedORIs.size() == 0)
		{
			removedORIs = new HashSet<IObjRef>();
		}
		removedORIs.add(objRef);
	}

	public void removeObjRefs(IObjRef[] objRefs)
	{
		for (IObjRef objRef : objRefs)
		{
			removeObjRef(objRef);
		}
	}

	public void removeObjRefs(List<IObjRef> objRefs)
	{
		for (int a = objRefs.size(); a-- > 0;)
		{
			removeObjRef(objRefs.get(a));
		}
	}

	public void removeObjRefs(Collection<IObjRef> objRefs)
	{
		for (IObjRef objRef : objRefs)
		{
			removeObjRef(objRef);
		}
	}

	public IRelationUpdateItem buildRUI()
	{
		RelationUpdateItem rui = new RelationUpdateItem();
		rui.setMemberName(memberName);
		rui.setAddedORIs(getAddedORIs());
		rui.setRemovedORIs(getRemovedORIs());
		return rui;
	}
}

File: ambeth/Ambeth.Merge/ambeth/merge/model/RelationUpdateItemBuild.cs
using De.Osthus.Ambeth.Collections;
using De.Osthus.Ambeth.Merge.Transfer;
using System;
using System.Collections;
using System.Collections.Generic;
namespace De.Osthus.Ambeth.Merge.Model
{
    public class RelationUpdateItemBuild : IRelationUpdateItem
    {
        protected String memberName;

        protected IISet<IObjRef> addedORIs = EmptySet.Empty<IObjRef>();

        protected IISet<IObjRef> removedORIs = EmptySet.Empty<IObjRef>();

        public RelationUpdateItemBuild(String memberName)
        {
            this.memberName = memberName;
        }

        public String MemberName
        {
            get
            {
                return memberName;
            }
        }

        public IObjRef[] AddedORIs
        {
            get
            {
                if (((ICollection)addedORIs).Count == 0)
                {
                    return null;
                }
                return addedORIs.ToArray();
            }
        }

        public IObjRef[] RemovedORIs
        {
            get
            {
                if (((ICollection)removedORIs).Count == 0)
                {
                    return null;
                }
                return removedORIs.ToArray();
            }
        }

        public void AddObjRef(IObjRef objRef)
        {
            if (((ICollection)addedORIs).Count == 0)
            {
                addedORIs = new CHashSet<IObjRef>();
            }
            addedORIs.Add(objRef);
        }

        public void AddObjRefs(IObjRef[] objRefs)
        {
            foreach (IObjRef objRef in objRefs)
            {
                AddObjRef(objRef);
            }
        }

        public void AddObjRefs(List<IObjRef> objRefs)
        {
            for (int a = 0, size = objRefs.Count; a < size; a++)
            {
                AddObjRef(objRefs[a]);
            }
        }

        public void AddObjRefs(IEnumerable<IObjRef> objRefs)
        {
            foreach (IObjRef objRef in objRefs)
            {
                AddObjRef(objRef);
            }
        }

        public void RemoveObjRef(IObjRef objRef)
        {
            if (((ICollection)removedORIs).Count == 0)
            {
                removedORIs = new CHashSet<IObjRef>();
            }
            removedORIs.Add(objRef);
        }

        public void RemoveObjRefs(IObjRef[] objRefs)
        {
            foreach (IObjRef objRef in objRefs)
            {
                RemoveObjRef(objRef);
            }
        }

        public void RemoveObjRefs(List<IObjRef> objRefs)
        {
            for (int a = objRefs.Count; a-- > 0; )
            {
                RemoveObjRef(objRefs[a]);
            }
        }

        public void RemoveObjRefs(IEnumerable<IObjRef> objRefs)
        {
            foreach (IObjRef objRef in objRefs)
            {
                RemoveObjRef(objRef);
            }
        }

        public IRelationUpdateItem BuildRUI()
        {
            RelationUpdateItem rui = new RelationUpdateItem();
            rui.MemberName = memberName;
            rui.AddedORIs = AddedORIs;
            rui.RemovedORIs = RemovedORIs;
            return rui;
        }
    }
}
